applications/vanilla/controllers/class.postcontroller.php

```diff
@@ -529,6 +530,12 @@ public function comment($DiscussionID = '') {
             $this->Form->addError(t('Failed to find discussion for commenting.'));
         }
 
+        // Vanilla Comments save as Text, no matter the format.
+        // Kludge to set format to Text if we're Wysiwyg to preserve newlines.
+        if ($isEmbeddedComments && ($this->Form->getFormValue('Format', c('Garden.InputFormatter')) === 'Wysiwyg')) {
```

Comment:
You say it saves as next no matter what, but only override if it's `Wysiwyg` - so which is it?